### ISSUE TITLE
Provide mock-ability for exceptions in testing

### ISSUE THREAD

**Issue by stweedie** [recurly/recurly-client-net#348]
Trying to test something similar to the following:
```c#
public bool AccountExists(string accountCode)
{
    try
    {
        var account = _recurlyAccountService.GetAccount(accountCode);
        return account != null;
    }
    catch (NotFoundException)
    {
        return false;
    }
}
```

but am currently unable to do so because I can't mock the NotFoundException.

Can you make the recurly exceptions have protected constructors instead of internal ones?

**Comment by aaron-suarez** [comment]
Hey there, @stweedie! Thanks for pointing this out. 

If you would please open a PR, we will surely merge it in for you! If you change the access modifier for one constructor, please keep it uniform by making all of the exceptions public.

All of the exceptions are here:

https://github.com/recurly/recurly-client-net/blob/master/Library/Exception.cs
https://github.com/recurly/recurly-client-net/blob/master/Library/InvalidCredentialsException.cs
https://github.com/recurly/recurly-client-net/blob/master/Library/NotFoundException.cs
https://github.com/recurly/recurly-client-net/blob/master/Library/ServerException.cs
https://github.com/recurly/recurly-client-net/blob/master/Library/TemporarilyUnavailableException.cs
https://github.com/recurly/recurly-client-net/blob/master/Library/TransactionError.cs
https://github.com/recurly/recurly-client-net/blob/master/Library/ValidationException.cs

Most of them use the `Error` class in their constructor, so it might be necessary to change the access modifiers on the constructors of classes as well, so that people can create the errors and pass them into the constructors of the exceptions

https://github.com/recurly/recurly-client-net/blob/master/Library/Error.cs
https://github.com/recurly/recurly-client-net/blob/master/Library/Errors.cs

Thank you in advance for your contribution to the library, I believe this will greatly improve the developer experience for those using this client.

**Comment by aaron-suarez** [comment]
Closed by #350
